src/Commands/new.js

```diff
@@ -2,88 +2,247 @@
+
+    if (options['skip-install'] === null) {
+      yield this._installDependencies()
+    }
+
+    yield this._cleanProjectDirectory()
+    yield this._copyEnvironmentFile()
+
+    if (options['skip-install'] === null) {
+      yield this._generateSecureKey()
+    }
+
+    this.log()
+    this.success(`${this.icon('success')} Your application is ready!`)
+    this.log()
+
+    this.info(`${this.icon('info')} Follow below instructions to get started`)
+    this.log(`$ cd ${this.colors.magenta.bold(args.name)}`)
+    this.log(`$ npm run serve:dev`)
+    this.log()
+  }
 
-const repo = 'https://github.com/adonisjs/adonis-app.git'
-let branch = 'master'
+  /**
+   * Verify that the application folder doesn't already exists.
+   *
+   * @private
+   * @method _verifyApplicationDoesntExist
+   * @return {void}
+   */
+  * _verifyApplicationDoesntExist () {
+    try {
+      yield pify(fs.access)(this.applicationPath, fs.constants.F_OK)
+      this.error(`${this.icon('error')} The directory "${this.applicationName}" already exists!`)
+      process.exit(0)
+    } catch (e) {
+      if (e.code !== 'ENOENT') {
+          this.error(`${this.icon('error')} An error occured while trying to access "${this.applicationPath}" directory`)
+          this.error(e.message)
+          process.exit(0)
+      }
+    }
+  }
 
-module.exports = function (argv) {
   /**
-   * switch to develop branch if --dev option is passed
-   * in commandline
+   * Clone the AdonisJs blueprint repository with
+   * the given branch.
+   *
+   * @private
+   * @method _cloneRepository
+   * @return {void}
    */
-  if (argv.dev) {
-    branch = 'develop'
+  * _cloneRepository () {
+    const branch = this.colors.magenta.bold(this.branch)
+    const blueprint = this.colors.magenta.bold(this.blueprint)
+
+    this._startSpinner(
+      this.colors.blue(`Cloning ${branch} branch of ${blueprint} blueprint`)
+    )
+
+    try {
+      yield pify(exec)(`git clone -b ${this.branch} --single-branch https://github.com/${this.blueprint}.git ${this.applicationPath}`)
+      this._stopSpinner()
+      this.log()
+      this.completed('clone', 'Repository cloned')
+    } catch (e) {
+      this.log()
+      this.error(`${this.icon('error')} An error occured while trying to clone ${branch} version of ${blueprint} blueprint.`)
+      process.exit(0)
+    }
   }
 
   /**
-   * return if project path is not defined
+   * Install all dependencies of the application.
+   *
+   * @private
+   * @method _installDependencies
+   * @return {void}
    */
-  if (!argv._[1]) {
-    console.log(colors.red(`Define project path \n${colors.bold.white('example:- adonis new yardstick')}`))
-    return
+  * _installDependencies () {
+    let tool = 'npm'
+
+    if (!this.mustUse && (yield this._hasYarnInstalled())) {
+      this.info(`${this.icon('info')} Yarn has been detected in your system!`)
+      if (yield this.confirm('Do you want to use it instead of npm?', false).print()) {
```

Comment:
Is it possible to write "Do you want to use yarn instead of npm?" instead?

To be honest, when I use a cli, I only read the question, and when I saw "it", I had to do an extra effort to read the line above.

I'm wondering about the default choice as well, should it be true by default @thetutlage ?